spec/meilisearch/index/base_spec.rb

```diff
@@ -126,6 +126,34 @@
     index.fetch_info
   end
 
+  it 'supports client_agents' do
+    custom_agent = 'Meilisearch Rails (v0.0.1)'
+    options = { timeout: 2, max_retries: 1, client_agents: [custom_agent] }
+    expected_headers = {
+      'Authorization' => "Bearer #{MASTER_KEY}",
+      'User-Agent' => "#{custom_agent};#{MeiliSearch.qualified_version}"
+    }
+
+    new_client = MeiliSearch::Client.new(URL, MASTER_KEY, options)
+    new_client.create_index!('options')
+    index = new_client.fetch_index('options')
+    expect(index.options).to eq(options.merge({ convert_body?: true }))
+
+    expect(MeiliSearch::Index).to receive(:get).with(
+      "#{URL}/indexes/options",
```

Comment:
A bit confusing to call the index `options`, it looks like a route name and not a resource `indexes/options`. 
But a part for that the test is good!

Reply:
Agreed, I'll generate an issue to handle that plus a way to randomize the indexes names as we do in the rails :)

Reply:
Issue created! #355